### PR TITLE
feat: create missing toasts

### LINKED ISSUE #150: Missing toasts
Every successful action and error should have an appropriate toast.

### PR DESCRIPTION
**Pull request by mdrutel**
#### Which issue(s) this PR fixes

Fixes #150 

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by PaulFarault** [comment]
Seems good.

Though, response objects are difficult to manipulate because tdp-server doesn't provide any schema for the return type of deploy endpoints. Thus, the OpenAPI Generator assigns an `any` type to the res object.

It may be worth fixing this issue (https://github.com/TOSIT-IO/tdp-server/issues/105) and updating the schema to benefit from Typescript for this PR.

**Review by PaulFarault — CHANGES_REQUESTED** [review]
@mdrutel well done for https://github.com/TOSIT-IO/tdp-server/issues/105

If you generate the client using the new schema, you'll see that we don't need to parse the response to JSON.

File: src/components/Services/VariablesDisplay/ValidateBar.tsx [hunk 1]
```diff
@@ -10,6 +11,7 @@ export function ValidateBar() {
     event.preventDefault()
     sendVariables(validateMessage)
     setValidateMessage('')
+    toast.info('Validating configuration...')
```
Comment (PaulFarault) [inline src/components/Services/VariablesDisplay/ValidateBar.tsx:14]: Is this toast is useful as the operation is almost instant?

File: src/pages/deploy/new/reconfigure.tsx [hunk 2]
```diff
@@ -42,7 +43,7 @@ export default function ReconfigurePage() {
         <Button
           as="button"
           variant={'filled'}
-          onClick={() => reconfigureDeploy()}
+          onClick={() => toast.info('Reconfiguring...') && reconfigureDeploy()}
```
Comment (PaulFarault) [inline src/pages/deploy/new/reconfigure.tsx:46]: Ideally, we should have a toast similar to the one for DAG and operations deployments. This one may not be useful without any id returned by the server.

**Review by PaulFarault — CHANGES_REQUESTED** [review]

**Comment by mdrutel** [comment]
I took into account all the comments above, after doing a rebase.

File: src/pages/deploy/new/reconfigure.tsx [hunk 3]
```diff
@@ -22,6 +23,19 @@ export default function ReconfigurePage() {
     fetchPreview()
   }, [planDeployReconfigure])
 
+  async function reconfigureLaunch() {
+    const res = reconfigureDeploy()
```
Comment (PaulFarault) [inline src/pages/deploy/new/reconfigure.tsx:27]: You forgot to remove this line:
```suggestion
```

File: src/pages/deploy/new/reconfigure.tsx [hunk 4]
```diff
@@ -22,6 +23,19 @@ export default function ReconfigurePage() {
...
+    try {
+      const res = await reconfigureDeploy()
+      console.log(res)
```
Comment (PaulFarault) [inline src/pages/deploy/new/reconfigure.tsx:30]: This one too:

```suggestion
```

File: src/pages/deploy/new/reconfigure.tsx [hunk 5]
```diff
@@ -22,6 +23,19 @@ export default function ReconfigurePage() {
...
+      res && toast.info(`Deploy id: ${res.id}`)
+    } catch (error) {
+      error.status &&
+        toast.error(`Error ${error.status} : ${error.statusText.toLowerCase()}`)
+      error.status === undefined && toast.error(String(error))
```
Comment (PaulFarault) [inline src/pages/deploy/new/reconfigure.tsx:35]: Won't it be better to use an `if`/`else` block here?

**Review by PaulFarault — CHANGES_REQUESTED** [review]

**Review by PaulFarault — APPROVED** [review]

**Review by pvalentin01 — APPROVED** [review]
Great,
Perhaps add missing toasts when you choose deployment from dag "from sources" or "to targets" and there's no operation to deploy
And also to reconfigure cluster if there's nothing to "deploy reconfigure"

**Comment by PaulFarault** [comment]
> Perhaps add missing toasts when you choose deployment from dag "from sources" or "to targets" and there's no operation to deploy

I'll open a new issue

**Pull request merged**